datafusion/expr/src/utils.rs

```diff
@@ -705,27 +711,20 @@ pub fn exprlist_to_fields<'a>(
         .map(|e| match e {
             Expr::Wildcard { qualifier, options } => match qualifier {
```

Comment:
Although we have moved wildcard expansions to the analyzer https://github.com/apache/datafusion/pull/11681, it still does wildcard expansions when computing plan schemas(in [exprlist_to_fields](https://github.com/apache/datafusion/blob/f9cc3325cdb5891b7566a6f3503c1f7ac6ad51e0/datafusion/expr/src/utils.rs#L696) and [exprlist_len](https://github.com/apache/datafusion/blob/f9cc3325cdb5891b7566a6f3503c1f7ac6ad51e0/datafusion/expr/src/utils.rs#L797C8-L797C20)). I wonder if performing wildcard expansions before computing schemas would be simplier, at least it would avoid redundant work.

Reply:
Another issue is that in [exprlist_to_fields](https://github.com/apache/datafusion/blob/f9cc3325cdb5891b7566a6f3503c1f7ac6ad51e0/datafusion/expr/src/utils.rs#L696), we don't handle replace items, but `ExpandWildcardRule` does.
`select * replace ('foo' as a) from t` will give the wrong datatype in schema before executing analyzer.